packages/react-router/src/router.ts

```diff
@@ -1534,9 +1535,10 @@ export class Router<
   }
 
   invalidate = () => {
-    const invalidate = (d: any) => ({
+    const invalidate = (d: RouteMatch<TRouteTree>) => ({
       ...d,
       invalid: true,
+      ...(d.status === 'error' ? ({ status: 'pending' } as const) : {}),
```

Comment:
some questions here @tannerlinsley:

- is there anything else we need to reset - like the `error` itself back to `undefined` ?
- do we need to do the reset conditionally? what if the route was in a state where loading data worked once and then error'd on a refetch?